addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -1846,3 +1846,73 @@ def test_check_lastcall_change_regular_to_accrual(self):
+                'name': 'Accrual Plan For Test',
+                'accrued_gain_time': 'start',
+                'carryover_date': 'year_start',
+                'level_ids': [(0, 0, {
+                    'start_count': 1,
+                    'start_type': 'day',
+                    'added_value': 1,
+                    'added_value_type': 'day',
+                    'frequency': 'daily',
+                    'cap_accrued_time': True,
+                    'maximum_leave': 10
+                })],
+            })
+            # Simulate creating an allocation from frontend interface
+            new_allocation = self.env['hr.leave.allocation'].new({
+                'name': 'Accrual allocation for employee',
+                'accrual_plan_id': accrual_plan.id,
+                'employee_id': self.employee_emp.id,
+                'holiday_status_id': leave_type.id,
+                'date_from': '2024-02-01',
+                'allocation_type': 'accrual',
+                'holiday_type': 'employee',
+            })
+            new_allocation._onchange_date_from()
+            record_data = new_allocation._convert_to_write(new_allocation._cache)
+            allocation = self.env['hr.leave.allocation'].create(record_data)
+            allocation.action_validate()
+            self.assertEqual(get_reamining_leaves(2024, 3, 2), 10,
+                             "The cap is reached, no more leaves should be accrued")
+
+            leave = self.env['hr.leave'].create({
+                'name': 'leave',
+                'employee_id': self.employee_emp.id,
+                'holiday_status_id': leave_type.id,
+                'request_date_from': '2024-02-25',
+                'request_date_to': '2024-03-01',
+            })
+            leave.action_validate()
+            self.assertEqual(get_reamining_leaves(2024, 3, 1), 5, "1 day should be deduced from the allocation")
+            self.assertEqual(get_reamining_leaves(2024, 3, 2), 6, "1 days should be added to the accrual allocation")
+            self.assertEqual(get_reamining_leaves(2024, 3, 2), 6, "Function return result should persist")
+            self.assertEqual(get_reamining_leaves(2024, 3, 10), 10, "Accrual allocation should be capped at 10")
+
+            leave = self.env['hr.leave'].create({
+                'name': 'leave',
+                'employee_id': self.employee_emp.id,
+                'holiday_status_id': leave_type.id,
+                'request_date_from': '2024-03-10',
+                'request_date_to': '2024-03-15',
+            })
+            leave.action_validate()
+            self.assertEqual(get_reamining_leaves(2024, 3, 9), 10, "Leaves following the date selected should not be counted")
+            self.assertEqual(get_reamining_leaves(2024, 3, 10), 5, "5 days should be deduced from the allocation")
+            self.assertEqual(get_reamining_leaves(2024, 3, 15), 5, "No bonus should be added during a leave")
```

Comment:
If the period at which the days are accrued is "start", we shouldn't take into account anything upon accruing days. The whole point of that is to not know what is coming :) So in this case, some days should have been accrued again after the 10th, because the balance has been impacted by the leave having started